src/Analyzers/Analyzers/src/StartupAnalyzer.cs

```diff
@@ -35,10 +36,13 @@ private void OnCompilationStart(CompilationStartAnalysisContext context)
                 return;
             }
 
+            var main = context.Compilation.GetEntryPoint(default)?.ContainingType;
+            var mainName = context.Compilation.GetEntryPoint(default)?.Name;
```

Comment:
```suggestion
            var entryPoint = context.Compilation.GetEntryPoint(context.CancellationToken);
            var main = entryPoint?.ContainingType;
            var mainName = entryPoint?.Name;
```

Reply:
You could remove the locals and just keep `entryPoint` since it has the most expressive name.